File: csharp/raindrops/Raindrops.cs
using System;

public static class Raindrops
{
    public static string Convert(int number)
    {
        string sound = "";
        if (number % 3 == 0) sound += "Pling";
        if (number % 5 == 0) sound += "Plang";
        if (number % 7 == 0) sound += "Plong";
        
        return (sound.Length == 0) ? number.ToString() : sound;
    }
}

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {

    fun convert(n: Int): String {
        var result = ""

        val numToSound = mapOf(3 to "Pling",
            5 to "Plang",
            7 to "Plong")

        for (key in numToSound.keys) {
            if (n.mod(key) == 0) {
                result = result.plus(numToSound[key])
            }
        }

        if (result.isEmpty()) {
            result = n.toString()
        }
        return result
    }
}
